### PR TITLE
PAYARA-3314 Docker Nodes Documentation

### PR DESCRIPTION
**Pull request by Pandrex247**
First cut of Docker Nodes documentation

### CODE REVIEW

File: documentation/payara-server/docker/docker-nodes.adoc [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+As Docker must expose an HTTP port, it is recommended that it be protected using TLS outside of developement environments.
+
+A guide for how to do so can be found here: https://docs.docker.com/engine/security/https/
+
+Assuming you followed this guide, you will need to perform the following steps to allow Payara Server to communicate
+with the Docker engine:
+
+1. Import the CA certificate into the Domain truststore
+2. Create a PKCS12 bundle from the client certificate and key
+3. Import the generated PKCS12 bundle into the Domain's keystore
```
Comment (rdebusscher) [inline documentation/payara-server/docker/docker-nodes.adoc]: indicate file of this keystore: $payara-home$/glassfish/domains/$domain-name$/config/keystore.jks

File: documentation/payara-server/docker/README.adoc [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[[docker-integration]]
+= Docker Integration
+
+Introduced in Payara Server 5.192 is the concept of a Docker node and instance.
+These function in a similar manner to instances hosted on a SSH, DCOM,
+or CONFIG node type, though as the name suggests are targetted towards creating
+and using instances housed within Docker containers.
+
```
Comment (rdebusscher) [inline documentation/payara-server/docker/README.adoc]: Indication of Technical Preview?

File: documentation/payara-server/docker/docker-instances.adoc [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+----
+asadmin create-instance --node DockerNode1 dockerInstance1
+----
+
+When you invoke this command, a Docker container of the same name will be created. If a Docker container of the same
+name already exists, the create-instance command will fail, and Payara Server will attempt to unregister the instance
+from the DAS.
+
+Please note that it is required that secure-admin be enabled for Docker instances to start (which is why a passwordfile
+is mandatory).
```
Comment (rdebusscher) [inline documentation/payara-server/docker/docker-instances.adoc]: which is why a passwordfile is mandatory _when we created the Docker Node_

File: documentation/payara-server/docker/docker-nodes.adoc [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+A Docker node is similar to the existing SSH, DCOM, and CONFIG nodes in that
+it is a virtual representation of a Payara Server installation, providing connection
+details that the Domain Administration Server (DAS) uses to communicate with the
+installation to create, stop, start, and delete instances. Where it
+differs from these existing node types is that instances created against it are actually housed in Docker containers,
+and so also contains details on how to communicate with the
+Docker engine of the remote machine.
+
+IMPORTANT: Docker must expose an HTTP port
```
Comment (OndroMih) [inline documentation/payara-server/docker/docker-nodes.adoc]: What is "Docker"? - a docker deamon, docker engine, docker container, docker image? Which HTTP port?

If this means that a Docker daemon must expose its REST admin service on an HTTP port, it should be clarified.

**Pull request merged**